File: backend/app/schemas/requirement.py
"""
功能点（需求）相关Schema
"""
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class RequirementBase(BaseModel):
    """功能点基础Schema"""
    requirement_name: Optional[str] = Field(None, max_length=200, description="需求名称")
    name: str = Field(..., max_length=500, description="功能点名称")
    description: Optional[str] = Field(None, description="功能点描述")
    category: Optional[str] = Field(None, max_length=50, description="类别")
    module: Optional[str] = Field(None, max_length=100, description="所属模块")
    priority: Optional[str] = Field(None, max_length=10, description="优先级")
    acceptance_criteria: Optional[str] = Field(None, description="验收标准")
    keywords: Optional[str] = Field(None, max_length=500, description="关键词")


class RequirementCreate(RequirementBase):
    """创建功能点Schema"""
    project_id: Optional[int] = Field(None, description="项目ID")
    version_id: Optional[int] = Field(None, description="版本ID")


class RequirementUpdate(BaseModel):
    """更新功能点Schema"""
    requirement_name: Optional[str] = Field(None, max_length=200, description="需求名称")
    name: Optional[str] = Field(None, max_length=500, description="功能点名称")
    description: Optional[str] = Field(None, description="功能点描述")
    category: Optional[str] = Field(None, max_length=50, description="类别")
    module: Optional[str] = Field(None, max_length=100, description="所属模块")
    priority: Optional[str] = Field(None, max_length=10, description="优先级")
    acceptance_criteria: Optional[str] = Field(None, description="验收标准")
    keywords: Optional[str] = Field(None, max_length=500, description="关键词")


class RequirementResponse(RequirementBase):
    """功能点响应Schema"""
    id: int
    project_id: Optional[int]
    version_id: Optional[int]
    created_at: datetime
    updated_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True


class RequirementListResponse(BaseModel):
    """功能点列表响应"""
    total: int
    items: List[RequirementResponse]


class RequirementGroupResponse(BaseModel):
    """需求分组响应Schema"""
    requirement_name: Optional[str]
    project_id: Optional[int]
    version_id: Optional[int]
    count: int
    created_at: datetime

File: frontend/src/types/requirement.ts
/**
 * 功能点（需求）相关类型定义
 */
export interface Requirement {
  id: number
  project_id: number
  version_id: number | null
  name: string
  description: string | null
  category: string | null
  module: string | null
  priority: string | null
  acceptance_criteria: string | null
  keywords: string | null
  created_at: string
  updated_at: string | null
}

export interface RequirementCreate {
  name: string
  project_id: number
  version_id?: number
  description?: string
  category?: string
  module?: string
  priority?: string
  acceptance_criteria?: string
  keywords?: string
}

export interface RequirementUpdate {
  name?: string
  description?: string
  category?: string
  module?: string
  priority?: string
  acceptance_criteria?: string
  keywords?: string
}

export interface RequirementListResponse {
  total: number
  items: Requirement[]
}
